src/wms/src/test/java/org/geoserver/wms/map/GetMapKvpRequestReaderTest.java

```
@@ -771,6 +772,65 @@ protected CloseableHttpResponse executeRequest(HttpCacheContext cacheContext, Ht
+        Header[] headers = new Header[1];
+        when(spyRequest.getHttpRequestHeader(authHeader)).thenReturn(fakeBasicAuth);
+
+        @SuppressWarnings("PMD.CloseResource") // mocked so no need for closing
+        CloseableHttpClient client = mock(CloseableHttpClient.class);
+        CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+        HttpEntity entity = mock(HttpEntity.class);
+
+        GetMapKvpRequestReader reqReader;
+        try (InputStream sld = GetMapKvpRequestReader.class.getResourceAsStream("BasicPolygonsLibraryNoDefault.sld")) {
+            when(entity.getContent()).thenReturn(sld);
+            when(response.getEntity()).thenReturn(entity);
+            when(client.execute(any(HttpGet.class), any(HttpCacheContext.class)))
+                    .thenReturn(response);
+
+            reqReader = new GetMapKvpRequestReader(wms, null) {
+                @Override
+                protected CloseableHttpClient createHttpClient(RequestConfig config, CacheConfiguration cacheConfig) {
+                    return client;
+                }
+            };
+        }
+
+        int numberOfThreads = 10;
+        ExecutorService service = Executors.newFixedThreadPool(numberOfThreads);
+        CountDownLatch latch = new CountDownLatch(numberOfThreads);
+        for (int i = 0; i < numberOfThreads; i++) {
+            service.submit(() -> {
+                try {
+                    URL url = new URL("http://localhost/geoserver/rest/sldurl/sample.sld");
+                    String urlDecoded = URLDecoder.decode(url.toExternalForm(), "UTF-8");
+                    HashMap kvp = new HashMap<>();
+                    kvp.put("layers", getLayerId(BASIC_POLYGONS));
+                    kvp.put("sld", urlDecoded);
+
+                    reqReader.read(spyRequest, parseKvp(kvp), caseInsensitiveKvp(kvp));
+                    assertSameHeader(headers, expectedBasicHeader);
+                } catch (Exception ignored) {
```

I don't think the exception should be ignored, the test report states this exception happening due to race conditions:

```
21 Jan 12:21:02 WARN   [geoserver.ows] - Exception while getting SLD.
java.lang.NullPointerException
	at org.geoserver.wms.map.GetMapKvpRequestReader.executeRequest(GetMapKvpRequestReader.java:1001)
	at org.geoserver.wms.map.GetMapKvpRequestReader.getHttpInputStream(GetMapKvpRequestReader.java:933)
```

If you're ignoring the exception because the request is not complete enough to pass a full reqReader.read, add some extra bits instead, but make the test fail in case any exception is thrown during reading.

---

good remark. i addressed you comments, made sure the requests succeeded and assert that the same httpclient instance is used. the test fails if any exception occurs during reading